Applications/FileIO/CsvInterface.cpp

```diff
@@ -181,4 +183,69 @@ std::size_t CsvInterface::findColumn(std::string const& line, char delim, std::s
     return count;
 }
 
+void CsvInterface::addIndexVectorForWriting(std::size_t s)
+{
+    std::vector<int> idx_vec(s);
+    std::iota(idx_vec.begin(), idx_vec.end(), 0);
+    addVectorForWriting("Index", idx_vec);
+}
+
+bool CsvInterface::write()
+{
+    if (_data.empty())
+    {
+        ERR ("CsvInterface::write() - No data to write.");
+        return false;
+    }
+
+    std::size_t const vec_size (getVectorSize(0));
+    std::size_t const n_vecs (_data.size());
+    for (std::size_t i=1; i<n_vecs; ++i)
+    {
+        if (getVectorSize(i) != vec_size)
+        {
+            ERR ("CsvInterface::write() - Data vectors are not equally long.");
+            return false;
+        }
+    }
+
+    _out << _vec_names[0];
+    for (std::size_t i=1; i<n_vecs; ++i)
+        _out << "\t " << _vec_names[i];
+    _out << std::endl;
+
+    for (std::size_t j=1; j<vec_size; ++j)
+    {
+        writeValue(0,j);
+        for (std::size_t i=1; i<n_vecs; ++i)
+        {
+            _out << "\t ";
+            writeValue(i,j);
+        }
+        _out << std::endl;
+    }
+    return true;
+}
+
+std::size_t CsvInterface::getVectorSize(std::size_t idx) const
+{
+    if (_data[idx].type() == typeid(std::vector<std::string>))
```

Comment:
Only one casting/typeiding around. It's just a note: Annoying people in order to educate them. ^^